### PR TITLE
[14.0][mig] website_sale_suggestion_create_account

### PR DESCRIPTION
**Pull request by dsolanki-initos**
[14.0][mig] website sale suggestion create account

### CODE REVIEW

**Review by ivantodorovich — COMMENTED** [review]
Hmmm, this is coming from the previous version, but I don't like that much this `customize_show` hack to pass tests.

For once, `customize_show="True"` views should be named, to avoid these ugly menu items:
image

But in fact in this case it doesn't seem that the module actually needs to use customize_show views, if it weren't for the tests.

Do we still need this hack in `14.0`?

**Comment by dsolanki-initos** [comment]
Hello @ivantodorovich,

Yes, we don't require customize_show view in this module so I've dropped the test case.

**Comment by ivantodorovich** [comment]
Well, I'd rather keep the ugly hack rather than dropping tests. We should rather try to make them work ;-)
And if it turns out it's not possible to remove this hack, I guess it makes more sense to keep it so we can have tests

But IMO this hack should work even without `customize_show`. You can still activate/deactivate the view without it

**Review by claudiagn — APPROVED** [review]
eventhough I see the comments below, it works! functional review ok :+1:

**Comment by dsolanki-initos** [comment]
@ivantodorovich will you please re-review it?

**Review by ivantodorovich — APPROVED** [review]

**Comment by dsolanki-initos** [comment]
@OCA/e-commerce-maintainers is it possible to merge this PR since it is approved?

**Comment by dsolanki-initos** [comment]
Is it possible to get merged this PR?

File: website_sale_suggest_create_account/tests/test_shop_buy.py
Comment (chienandalu) [inline website_sale_suggest_create_account/tests/test_shop_buy.py]: Why these changes? You're falling apart the code from its comment

File: website_sale_suggest_create_account/tests/test_shop_buy.py
Comment (chienandalu) [inline website_sale_suggest_create_account/tests/test_shop_buy.py]: What's the difference?

File: website_sale_suggest_create_account/tests/test_shop_buy.py
Comment (chienandalu) [inline website_sale_suggest_create_account/tests/test_shop_buy.py]: Isn't this the whole point of the test? :thinking:

**Review by chienandalu — COMMENTED** [review]
I don't understand some changes in the tests

**Review by chienandalu — COMMENTED** [review]
@Tardo Can you pay a review here when you can?

File: website_sale_suggest_create_account/tests/test_shop_buy.py
Comment (Tardo) [reply website_sale_suggest_create_account/tests/test_shop_buy.py]: Yes, this view must be enabled and complete the tour with all steps.

File: website_sale_suggest_create_account/tests/test_shop_buy.py
Comment (Tardo) [reply website_sale_suggest_create_account/tests/test_shop_buy.py]: I would say there is no difference

**Review by Tardo — COMMENTED** [review]
Functionally OK, but heed @chienandalu comments. Thanks.

**Comment by dsolanki-initos** [comment]
@chienandalu and @Tardo can you please re-review it.

**Comment by dsolanki-initos** [comment]
Is it possible to merge this PR?

File: website_sale_suggest_create_account/tests/test_shop_buy.py
Comment (chienandalu) [inline website_sale_suggest_create_account/tests/test_shop_buy.py]: This still makes no sense. You're supposed to test this module without being logged in, which is the point of the test

File: website_sale_suggest_create_account/static/tests/tours/checkout.js
Comment (Tardo) [inline website_sale_suggest_create_account/static/tests/tours/checkout.js:8]: Please, complete this tour...

**Review by Tardo — CHANGES_REQUESTED** [review]

**Comment by dsolanki-initos** [comment]
It would be great if we merge the PR.

**Comment by dsolanki-initos** [comment]
It would be great if we merge the PR.

**Comment by dsolanki-initos** [comment]
It would be nice if maintainers merge it.

**Comment by pedrobaeza** [comment]
There are pending comments.

File: website_sale_suggest_create_account/static/tests/tours/checkout.js
Comment (dsolanki-initos) [reply website_sale_suggest_create_account/static/tests/tours/checkout.js:8]: Tour is complete. All the steps are followed.Let me know the steps which is missed from my end

**Comment by dsolanki-initos** [comment]
It would be great if got approval from the reviewer.

**Comment by anajuaristi** [comment]
Pleas @tardo, could you give approval to @dsolanki-initos so they could merge it? Thank you!

**Comment by dsolanki-initos** [comment]
Hello, @Tardo ,@chienandalu it would be nice if both give approval so the maintainer could merge it ASAP.

**Comment by chienandalu** [comment]
@dsolanki-initos you didn't fix the test @Tardo asked you

**Comment by sebasdrk17** [comment]
Hello @dsolanki-initos, I've made the fixes mentioned in [here](https://github.com/OCA/e-commerce/pull/466/files/1b9f17cfd6ceda59be35e0d519ac123245631f61#r631223553), the step missing was the part where we check that the checkout button doesn't exists.

Can I push the changes?, this module is needed in order to migrate `website_sale_require_login`

**Comment by oscarolar** [comment]
@OSevangelist hey man  are you ok with this?

> Hello @dsolanki-initos, I've made the fixes mentioned in [here](https://github.com/OCA/e-commerce/pull/466/files/1b9f17cfd6ceda59be35e0d519ac123245631f61#r631223553), the step missing was the part where we check that the checkout button doesn't exists.
> 
> Can I push the changes?, this module is needed in order to migrate `website_sale_require_login`

**Comment by dsolanki-initos** [comment]
> @OSevangelist hey man are you ok with this?
> 
> > Hello @dsolanki-initos, I've made the fixes mentioned in [here](https://github.com/OCA/e-commerce/pull/466/files/1b9f17cfd6ceda59be35e0d519ac123245631f61#r631223553), the step missing was the part where we check that the checkout button doesn't exists.
> > Can I push the changes?, this module is needed in order to migrate `website_sale_require_login`

If it is urgent you can add the changes.

**Comment by dsolanki-initos** [comment]
@sebasdrk17 do you already started work on this module?

**Comment by dsolanki-initos** [comment]
Hi, @sebasdrk17 do you already started work on this module?

**Pull request closed without merging**
